drivers/digital-io/max14919/max14919.c

```diff
@@ -0,0 +1,204 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include "max14919.h"
+#include "no_os_alloc.h"
+
+/**
+ * @brief Set the OUT channels state
+ * @param desc - MAX14919 device descriptor
+ * @param state - OUT channels state array containing information about the to
+ * 		  be set channels state.
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int max14919_set_out(struct max14919_desc *desc, enum max14919_out_state *state)
+{
+	int ret, i;
+
+	for (i = 0; i < MAX14919_OUT_CHANNELS; i++) {
+		ret = no_os_gpio_set_value(desc->in_desc[i],
+					   state[i] ? NO_OS_GPIO_LOW : NO_OS_GPIO_HIGH);
+		if (ret)
+			return ret;
+
+		desc->out_state[i] = state[i];
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Set current limiting
+ * @param desc - MAX14919 devce descriptor
+ * @param state - State to be set.
+ * 		  TRUE - on.
+ * 		  FALSE - off.
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int max14919_set_climit(struct max14919_desc *desc, bool state)
+{
+	if (!desc->inrush_desc)
+		return -EINVAL;
+
+	return no_os_gpio_set_value(desc->inrush_desc,
+				    state ? NO_OS_GPIO_HIGH : NO_OS_GPIO_LOW);
+}
+
+/**
+ * @brief Get the FAULT state of the device
+ * @param desc - MAX14919 device descriptor
+ * @return 0 - a fault has occurred.
+ * 	   1 - no faults have been reported.
+*/
+int max14919_get_fault(struct max14919_desc *desc)
+{
+	uint8_t fault;
+	int ret;
+
+	ret = no_os_gpio_get_value(desc->fault_desc, &fault);
```

Comment:
check desc pointer before and return fault with pointer, we always return the status of the API execution with return, and any other values over parameters with pointers.
see everywhere where applicable

Reply:
this has not been resolved, please return fault value as pointer

Reply:
Solved it.